code/modules/mob/living/carbon/human/human_movement.dm

```diff
@@ -107,3 +107,54 @@
 	if(shoes && (shoes.item_flags & NOSLIP) && istype(shoes, /obj/item/clothing/shoes/magboots))  //magboots + dense_object = no floating
 		return 1
 	return 0
+
+/mob/living/carbon/human/handle_footstep(atom/T)
+	if(..())
+		if(T.footstep_sounds["human"])
+			var/S = pick(T.footstep_sounds["human"])
+			if(S)
+				if(m_intent == "run")
+					if(!(step_count % 2)) //every other turf makes a sound
+						return 0
+
+				var/range = -(world.view - 2)
+				if(m_intent == "walk")
+					range -= 0.333
+				if(!shoes)
+					range -= 0.333
+
+				//shoes + running
+					//-(7 - 2) = -(5) = -5 | -5 - 0           = -5     | (7 + -5)     = 2     | 2     * 3 = 6     | range(6)     = range(6)
+				//running OR shoes
+					//-(7 - 2) = (-5) = -5 | -5 - 0.333       = -5.333 | (7 + -5.333) = 1.667 | 1.667 * 3 = 5.001 | range(5.001) = range(5)
+				//walking AND no shoes
+					//-(7 - 2) = (-5) = -5 | -5 - (0.333 * 2) = -5.666 | (7 + -5.666) = 1.334 | 1.334 * 3 = 4.002 | range(4.002) = range(4)
+
+				var/volume = 90
+				if(m_intent == "walk")
+					volume -= 55
+				if(!shoes)
+					volume -= 70
+
+				if(istype(shoes, /obj/item/clothing/shoes))
+					var/obj/item/clothing/shoes/shooess = shoes
+					if(shooess.silence_steps)
+						return 0 //silent
+
+				if(!has_organ("l_foot") && !has_organ("r_foot"))
+					return 0 //no feet no footsteps
+
+				if(buckled || lying || throwing)
+					return 0 //people flying, lying down or sitting do not step
+
+				if(!has_gravity(src))
+					if(step_count % 3) //this basically says, every three moves make a noise
+						return 0       //1st - none, 1%3==1, 2nd - none, 2%3==2, 3rd - noise, 3%3==0
+
+				if(species.silent_steps)
+					return 0 //species is silent
+
+				playsound(T, S, volume, 1, range)
+				return 1
+
+	return 0
```

Comment:
Do not use return if there's no any actions in function after it. Do not return unused values from functions, use return without param if it acts like break.